File: models/computer.go
package models

// Computer contains computer data
type Computer struct {
	ID                 int     `json:"id"`
	Name               string  `json:"name"`
	Mac                string  `json:"mac"`
	IPAddress          string  `json:"ip"`
	BroadcastIPAddress string  `json:"broadcast"`
	SSHEnabled         bool    `json:"sshEnabled"`
	SSH                SSHInfo `json:"sshInfo"`
	ESXEnabled         bool    `json:"esxEnabled"`
	Esx                ESXInfo `json:"esxInfo"`
}

// SSHInfo contains ssh access info
type SSHInfo struct {
	Username     string `json:"username"`
	Password     string `json:"password"`
	UsePublicKey bool   `json:"usePublicKey"`
	PublicKey    string `json:"publicKey"`
}

// ESXInfo contains esx cached info
type ESXInfo struct {
	VMs []EsxVM `json:"vms"`
}

// EsxVM contains virtual host status
type EsxVM struct {
	ID     int    `json:"id"`
	Name   string `json:"name"`
}

File: ui/src/models/Computer.ts

export interface Computer {
  ID: string,
  Name: string,

  Status: Status,
  Child: ChildComputer[]
}

export type ChildComputer = Exclude<Computer, ["IPAddress", "Child"]>;

export type ComputerActionArgs = {
  computer: Computer,
  child?: ChildComputer,
  action: CommandAction,
}
export type CommandAction = "POWERON" | "SUSPEND" | "SHUTDOWN" | "EDIT";
export type Status = "" | "ONLINE" | "OFFLINE" | "PAUSED" | "STOPPED" | "RUNNING" | "PENDING"


export interface ComputerEditViewModel extends Computer {
  IPAddress: string,
  BroadcastIPAddress: string,
  MAC: string,
  SSHEnabled: boolean,
  isESXHost: boolean
}

interface SSHInfo {
  Username: string,
  Password: string,
  UsePublicKey: boolean,
  PublicKey: string,
}